### PR TITLE
docs: fix several missing/broken entries

### PR DESCRIPTION
**Pull request by shiftinv**
## Summary

- adds previously missing `application_command` and `command_failed` attributes to `ApplicationCommandInteraction` docs
- adds `qualified_name` to `InvokableApplicationCommand` (+ subclasses) docs
- adds `:inherited-members:` to `InteractionMessage`
- fixes missing `InstallParams` docs and broken `versionadded` directive
- adds `disnake.ext.commands.option_enum`

## Checklist

- [ ] If code changes were made, then they have been tested
    - [ ] I have updated the documentation to reflect the changes
    - [ ] I have formatted the code properly by running `task lint`
    - [ ] I have type-checked the code by running `task pyright`
- [ ] This PR fixes an issue
- [ ] This PR adds something new (e.g. new method or parameters)
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [x] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

**Comment by onerandomusername** [comment]
Please add `:inherited-members:` to `InteractionMessage` in api.rst

**Comment by shiftinv** [comment]
> Please add `:inherited-members:` to `InteractionMessage` in api.rst

38ed87f1219a181c8f726126d3358a3ed1422db7

File: disnake/ext/commands/slash_core.py [hunk 1]
```diff
@@ -109,6 +109,9 @@ class SubCommandGroup(InvokableApplicationCommand):
     ----------
     name: :class:`str`
         The name of the group.
+    qualified_name: :class:`str`
```
[the remark this answers is not in the document]
Comment (shiftinv) [reply disnake/ext/commands/slash_core.py:112]: Would be easier, but `:inherited-members:` doesn't support attributes defined this way

[the remark this answers is not in the document]
Comment (onerandomusername) [reply disnake/ext/commands/slash_core.py:112]: In that case, InteractionMessage needs to have them copied as well.

[the remark this answers is not in the document]
Comment (shiftinv) [reply disnake/ext/commands/slash_core.py:112]: good catch, fixed

File: disnake/ext/commands/params.py [hunk 2]
```diff
@@ -929,6 +929,7 @@ def inject(function: Callable[..., Any]) -> Any:
 def option_enum(
     choices: Union[Dict[str, TChoice], List[TChoice]], **kwargs: TChoice
 ) -> Type[TChoice]:
+    """A utility function to create an enum type with the provided values."""
```
Comment (onerandomusername) [inline disnake/ext/commands/params.py:932]: Should have return type and other directives marked. Which version was this added in? 
image

Comment (shiftinv) [reply disnake/ext/commands/params.py]: That method was added in 2.1 - updated the docs: https://github.com/DisnakeDev/disnake/pull/470/commits/1784630a21b3ed5fa17502bb7e25b84215562503

**Review by onerandomusername — APPROVED** [review]
lgtm

**Pull request merged**